src/later_shim.h

```diff
@@ -0,0 +1,7 @@
+#ifndef LATER_SHIM_H
+#define LATER_SHIM_H
+
+// This is simply a shim so that later::later can be accessed from C, not C++
```

Comment:
I really designed `later_api.h` for packages that use Rcpp, luckily this little shim is all that's needed to make it work with C. (If you're aware of a better technique than this, please let me know)

Reply:
I've switched to calling `execLaterNative2` in fb14833 which is what `later_api.h` appears to be doing anyway. It is called once in the package init function, so should be safe as well. Unless there are pitfalls I am not aware of, I prefer this method as it (i) utilises the 'official' R linking API and (ii) does not need a C++ compiler to build the `nanonext` shared object.

Reply:
I am now initializing the `GetCCallable` to `execLaterNative2`  when the first promise is 'registered' by calling back into R to load the `later` namespace. This is inspired by an approach Winston Chang and I took to 'lazy-load' `rlang` in the `later` package itself. This means that there is no implementation overhead now for cases that do not use promises.